.travis.yml

```diff
@@ -19,8 +19,10 @@ install:
   - rustup target add thumbv7em-none-eabi
   - rustup target add riscv32imc-unknown-none-elf
   - rustup component add rustfmt-preview
+  - rustup component add clippy
 
 script:
   - cargo fmt --all -- --check
   - cargo test --workspace
+  - cargo clippy --workspace --all-targets
```

Comment:
Every check we add to Travis that we cannot easily run by hand hurts development velocity. Travis takes a long time to give results, and bad pushes make the code review process more complex.

Reply:
If you're using vscode the lints appear as warning in your IDE. If we want to improve feedback cycles we can also remove the mac os target from travis which would save a lot of time.

Reply:
All three steps could be run on the command line. What about an additional script `check_commit.sh` that combines all those steps?

Reply:
I'm not using `vscode`, so those warnings don't help me. I use a text editor with code highlighting and minimal indentation support.

Travis can only be sped up so much, and it still requires you to push your (possibly-broken) revisions into PRs that are currently under review. Perhaps we can have a local script that runs these checks instead?

Reply:
> All three steps could be run on the command line. What about an additional script `check_commit.sh` that combines all those steps?

Ah, you responded while I was typing up my comment. I have the same idea.